html/semantics/forms/the-form-element/form-requestsubmit.html

```diff
@@ -0,0 +1,84 @@
+<!DOCTYPE html>
+<body>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<iframe name="iframe" src="about:blank"></iframe>
+
+<script>
+test(() => {
+  document.body.insertAdjacentHTML('afterbegin', '<form><input type="reset"></form>');
+  let form = document.querySelector('form');
+  assert_throws(new TypeError(), () => {
+    form.requestSubmit(document.body);
+  });
+  assert_throws(new TypeError(), () => {
+    form.requestSubmit(form.querySelector('input[type=reset]'));
+  });
+}, 'Passing an element which is not a submit button should throw');
+
+test(() => {
+  document.body.insertAdjacentHTML('afterbegin', '<form><input type="reset"></form>');
+  let form = document.querySelector('form');
+  let submitButton = document.createElement('button');
+  submitButton.type = 'submit';
+  assert_throws(new TypeError(), () => {
+    form.requestSubmit(submitButton);
+  });
+}, 'Passing a submit button not owned by the context object should throw');
```

Comment:
The element isn't even in the same tree, here. There are other interesting tests as well.

* The control is a descendant, but has a `form` attribute overriding the form owner to a different form.
* The control is a descendant, but has a `form` attribute that gives the control *no* form owner (e.g. `form=""`)
* The control is a descendant, but is actually owned by a different form by association from the parser. [1] or [2]

For each case, check that `A.requestSubmit(control)` throws, and `B.requestSubmit(control)` works.

[1] nested form actually owns the control
```
<!DOCTYPE html>
<form action=data:,A>
<div>
</form>
<form action=data:,B>
<input type=submit>
</form>
</div>
```
[2] the control is a descendant of the A form, not the B form, but is actually owned by B
```
<!DOCTYPE html>
<form action=data:,A>
<table>
</form>
<form action=data:,B>
<tr><td><input type=submit></td></tr>
</form>
</table>
```

Reply:
This is not a test of ```.form``` IDL attribute, and IMO  testing various patterns of ```.form``` here isn't cost-effective.  I added a few cases which ensures UAs don't apply simple ancestor check.